lib/puppet/type.rb

```diff
@@ -1966,7 +1966,7 @@ def self.unprovide(name)
   # @return [Array<Puppet::Provider>] Returns an array of all suitable providers.
   #
   def self.suitableprovider
-    providerloader.loadall if provider_hash.empty?
+    providerloader.loadall(Puppet.lookup(:current_environment)) if provider_hash.empty?
```

Comment:
`provider_hash` never seems to be empty, so this seems like a dead line

Reply:
Want to keep it anyway?

Reply:
Yes, I haven't been able come up with a case where it's needed (tried types without any providers, providers whose symbolic name doesn't match it's filename)... but hard to prove that it's never called. I'll leave it as is.